go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/argoproj/pkg
 
-go 1.14
+go 1.22.5
```

Comment:
Per the [Go docs](https://go.dev/doc/modules/gomod-ref#go), this is the _minimum_ required Go version. Afaik, this repo does not use any newer Go features and so does not need to raise its version

That is also different from the version this compiles with or is linted with, which is listed in the GHA Workflows, e.g. https://github.com/argoproj/pkg/blob/42ec18a0638911de9d98b8faaabf2decfdf883cd/.github/workflows/go.yml#L19

The consuming applications also set their own Go versions that compile this package

Reply:
sure, I reverted it